src/rapids_pre_commit_hooks/alpha_spec.py

```diff
@@ -0,0 +1,204 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from functools import reduce
+
+import yaml
+from packaging.requirements import Requirement
+from packaging.specifiers import SpecifierSet
+
+from .lint import LintMain
+
+RAPIDS_VERSIONED_PACKAGES = {
+    "rmm",
+    "pylibcugraphops",
+    "pylibcugraph",
+    "nx-cugraph",
+    "dask-cudf",
+    "cuspatial",
+    "cuproj",
+    "cuml",
+    "cugraph",
+    "cudf",
+    "ptxcompiler",
+    "cubinlinker",
+    "cugraph-dgl",
+    "cugraph-pyg",
+    "cugraph-equivariant",
+    "raft-dask",
+    "pylibwholegraph",
+    "pylibraft",
+    "cuxfilter",
+    "cucim",
+    "ucx-py",
+    "ucxx",
+    "pynvjitlink",
+    "distributed-ucxx",
+}
+
+ALPHA_SPECIFIER = ">=0.0.0a0"
+
+ALPHA_SPEC_OUTPUT_TYPES = {
+    "pyproject",
+    "requirements",
+}
+
+
+def check_package_spec(linter, args, node):
+    if node.tag == "tag:yaml.org,2002:str":
```

Comment:
I don't know the YAML spec very well to verify these tags. Happy to look into them a bit if it's helpful, but if someone else is already familiar (@bdice, @ajschmidt8, @jameslamb, ...) is already familiar that would be great.

Reply:
I'm seeing this pattern repeat a lot, however. Should we consider adding a helper function like 

```python
def _node_has_type(node, tag_type):
    return node.tag == f"tag:yaml.org,2002:{tag_type}"
```
?

Reply:
Done